### PR TITLE
Supporting map properties in struct via custom function

### PR DESCRIPTION
**Pull request by andrewcretin**
Any reason why the Map object type is not supported on custom functions

### CODE REVIEW

**Comment by brianvoe** [comment]
No specific reason maps were not added. Its always been an evolving process. Just kept tackling specific situations people throw my way.

File: struct.go [hunk 1]
```diff
@@ -234,7 +235,7 @@ func rSlice(ra *rand.Rand, t reflect.Type, v reflect.Value, tag string, size int
 	return nil
 }
 
-func rString(ra *rand.Rand, t reflect.Type, v reflect.Value, tag string) error {
+func rString(ra *rand.Rand, _ reflect.Type, v reflect.Value, tag string) error {
```
Comment (brianvoe) [inline struct.go:238]: If reflect type isnt being use you can just remove it in the switch statement call and not need to pass it.

File: struct.go [hunk 2]
```diff
@@ -322,7 +323,7 @@ func rFloat(ra *rand.Rand, t reflect.Type, v reflect.Value, tag string) error {
 	return nil
 }
 
-func rBool(ra *rand.Rand, t reflect.Type, v reflect.Value, tag string) error {
+func rBool(ra *rand.Rand, _ reflect.Type, v reflect.Value, tag string) error {
```
Comment (brianvoe) [inline struct.go:326]: Same as above

**Comment by brianvoe** [comment]
I think this looks good to me. Ill probably move the generator specific functions in the generate file just because that are not general helpers and are specific to just the generate functionality. I'll post a version update once I have it in. Thanks for this!!!

**Comment by andrewcretin** [comment]
Awesome, sounds good

**Comment by brianvoe** [comment]
v6.8.0

**Pull request merged**